iceoryx_posh/CMakeLists.txt

```diff
@@ -135,16 +135,17 @@ add_library(iceoryx_posh
     source/runtime/ipc_runtime_interface.cpp
     source/runtime/ipc_message.cpp
     source/runtime/port_config_info.cpp
-    source/runtime/posh_runtime.cpp
-    source/runtime/posh_runtime_impl.cpp
-    source/runtime/posh_runtime_single_process.cpp
+    source/runtime/posh_runtime.cpp                #
+    source/runtime/posh_runtime_impl.cpp           # @todo iox-#590 These files should go into a separate library iceoryx_posh_runtime
+    source/runtime/posh_runtime_single_process.cpp #
     source/runtime/node.cpp
```

Comment:
@budrus Why is the node in runtime? Does this have a specific reason?

Reply:
@mossmaurice Not really. I fear we just haven't found a better place. Feel free to suggest changes

Reply:
As just discussed, for me the node is part of a hierarchy described in [this PR](https://github.com/eclipse-iceoryx/iceoryx/pull/620/files#diff-7d0aab58cfe67a9cc98e780db81a4504ac49c4094ae5e2ba073a2b71caf91ba9R51). Let's tackle this in the follow-up PR.